### PR TITLE
Markdown: Escape some elements prior to sending them through Markdown

### PR DESCRIPTION
**Pull request by annando**
Fixes https://github.com/friendica/friendica-addons/pull/924#discussion_r361459823

### CODE REVIEW

File: markdown/markdown.php [hunk 1]
```diff
@@ -49,5 +49,25 @@ function markdown_post_local_start(App $a, &$request) {
...
+
 	$request['body'] = Markdown::toBBCode($request['body']);
+
+	foreach ($elements as $element) {
+		$request['body'] = preg_replace_callback("/\[base64" . $element . "\](.*?)\[\/base64" . $element . "\]/ism",
+					function ($match)  use ($element) {
+						return '[' . $element . ']' . base64_decode($match[1]) . '[/' . $element . ']';
+					},
+					$request['body']
+				);
```
Comment (MrPetovan) [inline markdown/markdown.php]: Code standards: please deindent this block twice.

File: markdown/markdown.php [hunk 2]
```diff
@@ -49,5 +49,25 @@ function markdown_post_local_start(App $a, &$request) {
...
 
+	// Elements that shouldn't be parsed
+	$elements = ['code', 'noparse', 'nobb', 'pre'];
+	foreach ($elements as $element) {
+		$request['body'] = preg_replace_callback("/\[" . $element . "\](.*?)\[\/" . $element . "\]/ism",
+					function ($match)  use ($element) {
+						return '[base64' . $element . ']' . base64_encode($match[1]) . '[/base64' . $element . ']';
+					},
+					$request['body']
+				);
```
Comment (MrPetovan) [inline markdown/markdown.php]: Code standards: please deindent this block twice.

File: markdown/markdown.php [hunk 3]
```diff
@@ -49,5 +49,25 @@ function markdown_post_local_start(App $a, &$request) {
...
+			function ($match)  use ($element) {
+				return '[base64' . $element . ']' . base64_encode($match[1]) . '[/base64' . $element . ']';
+			},
+			$request['body']
+		);
+	}
+
 	$request['body'] = Markdown::toBBCode($request['body']);
+
+	foreach (array_reverse($elements) as $element) {
```
Comment (MrPetovan) [inline markdown/markdown.php]: What is the point of this `array_reverse`?

Comment (annando) [reply markdown/markdown.php]: You could have nested elements. Then they have to be decoded in the opposite order.

**Pull request merged**